sabelotodo/routes.py

```diff
@@ -50,9 +50,41 @@ def create_item():
         db.session.rollback()
         return "Database error", 500
 
-    return jsonify(item), 200
-    # TODO: This should use item_schema.dump, but currently this leads to small
-    # discrepancies in date string format that break tests. (lbv 2020-4-2)
+    return item_schema.dumps(item), 200
+
+
+@app.route('/item/<int:itemid>', methods=["PATCH"])
+def patch_item(itemid):
+    item = Item.query.get_or_404(itemid)
+    json_data = request.get_json()
+
+    if not json_data:
+        return "No data provided", 400
+    if "id" in json_data:
+        return "ID numbers shouldn't change", 400
+
+    item_data = item_schema.dump(item)
+    item_data.update(json_data)
+
+    try:
+        item_schema.load(item_data)
+        for k, v in item_data.items():
+            setattr(item, k, v)
+    # We copy the attributes over like this rather than create a new item so
+    # that we're modifying the existing DB row rather than creating a duplicate
+    # one with a new ID.
+
+    except ValidationError as v:
+        return "JSON provided doesn't match schema when combined with specified item: %s" % v, 400
+
+    try:
+        db.session.commit()
+    except SQLAlchemyError:
+        # TODO: This branch of code is currently untested (lbv)
```

Comment:
ooooh What I would do here is mock db.session such that it throws the error, and then test that rollback gets called and the correct message is returned. Does that make sense? Should we take some time to talk mocking?

Reply:
(Open a ticket for it and we can do it as a separate PR.)